Quick set shooter

HarrisHamid opened the pull request:
**Purpose**

The purpose of this PR was to fix the QuickSetShooter setRedZone not responding issue and adjusted piston setting numbers. Also commented out double solenoid code that was causing errors.

**Description**
As a group, we fixed:

 setRedZone
- corrected issue where isFinished was set to false
commented out double solenoids
changed the module number, forward channel, and reverse channel for the pistons

**Testing**
Currently, this code builds and shooting and firing at different zones have been tested

bjmcternan commented:
@HarrisHamid This PR does a lot more than just fix the RED zone issue that we saw. This is a complete revamp of the firing mechanism and RPM control. There's also code for the piston manipulation.

Don't forget to mention how the XBox button mapping! Please edit the description

bjmcternan commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -109,52 +97,80 @@ private void configureButtonBindings()
     public Joystick getleftJoy() 
     {
         return leftJoy;
+
```
Extra whitespace, this is OK for now

bjmcternan commented on src/main/java/frc/robot/commands/Fire.java:
```diff
@@ -51,18 +54,27 @@ public void initialize()
+        {
+            //Not ready! Don't Feed!
+            m_feeder.feedOff();
+    	    m_feeder.beltOff();
```
Tab issue, this is OK for now

bjmcternan commented on src/main/java/frc/robot/commands/Fire.java:
```diff
@@ -51,18 +54,27 @@ public void initialize()
     public void end(boolean interrupted) 
     {
-    	m_feeder.feedOff();
+        m_feeder.feedOff();
```
Tab issue, this is ok for now

bjmcternan commented on src/main/java/frc/robot/commands/SetBlueZone.java:
```diff
@@ -78,4 +71,4 @@ public boolean runsWhenDisabled()
         // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
     }
-}
+}
```
This needs a whitespace at the end of the file

bjmcternan commented on src/main/java/frc/robot/commands/SetGreenZone.java:
```diff
@@ -0,0 +1,74 @@
+        return false;
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+    }
+}
```
This needs a white space at the end of the file

bjmcternan commented on src/main/java/frc/robot/commands/SetRedZone.java:
```diff
@@ -0,0 +1,68 @@
+        return true;
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+    }
+}
```
This needs a whitepace at the end of the file

bjmcternan commented on src/main/java/frc/robot/commands/SetYellowZone.java:
```diff
@@ -0,0 +1,67 @@
+        return true;
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DISABLED
+    }
+}
```
This needs a whitespace at the end of the file

bjmcternan commented on src/main/java/frc/robot/subsystems/Launcher.java:
```diff
@@ -88,56 +76,76 @@ public void simulationPeriodic()
     }
 
-    public void enableLauncher(double rpm) 
+    //Converys Encoder units per desiseconds (100ms) to RPMs
```
Misspelling, converts

bjmcternan commented on src/main/java/frc/robot/subsystems/LauncherController.java:
```diff
@@ -0,0 +1,138 @@
+
+
+    @Override
+    public void periodic() {
```
Fix the {}'s

bjmcternan commented on src/main/java/frc/robot/subsystems/LauncherController.java:
```diff
@@ -0,0 +1,138 @@
+        m_launcherState = LauncherState.OFF;
+        m_launcher = launcher;
+    }
+
```
Extra whitespace

bjmcternan commented on src/main/java/frc/robot/subsystems/LauncherController.java:
```diff
@@ -0,0 +1,138 @@
+    @Override
+    public void simulationPeriodic() {
+        // This method will be called once per scheduler run when in simulation
+
```
Extra whitespace

bjmcternan reviewed (changes requested):
There's some whitespace issues that needs to be addressed. While you're in there you might as well fix the TAB issues as well.

HarrisHamid commented:
**Purpose**

The purpose of this PR was to create QuickSetShooter buttons on an Xbox controller. When a button is pressed, the motor connected to the shooter wheel will set the  RPM to the RPM mapped to that button.  Depending on the button pushed, either the speed would change or turn the motor off.  A fire button was also included to fire balls when held, which would also manipulate the launcher and feeder. Lastly, shooterpiston buttons were added in which the d-pad on an Xbox controller would allow it to extend or retract.

**Description**

As a group, we:

setShootingZones
- The XYBA button on an Xbox controller would have a certain RPM mapped depending on which zone they were assigned. A click of a button once would set the RPM to the specific speed, the same button again would turn the motor off, and a different button clicked would change the overall speed.
- An issue resolved was fixing the RedZone, which was not responding/ setting to the mapped RPM, by setting isFisnihed to true and correctly assigning the buttons in robotcontainer.
Fire Button
- The fire button was mapped to a bumper and would need to be held in order to fire since it controlled the feeder and launcher. Once it was let go, the feeder and launcher would disable, but the motor controlling the wheel would continue to spin.
- corrected issue where isFinished was set to false
ShooterPiston
- Controls the piston meant to manipulate the launcher angle for the green zone. The d-pad is used; when pushed up, it should extend the piston, while pushing it down should retract it.

-commented out double solenoids
-changed the module number, forward channel, and reverse channel for the pistons

**Testing**
Currently, this code builds, and shooting and firing at different zones has been tested. 
The shooter piston however is not operational with the d-pad.

bjmcternan reviewed (approved):
Looks good. There are some minor issues but we can get them on the next PR.
